Source/WebGPU/WebGPU/BindGroupLayout.mm

```diff
@@ -232,11 +231,18 @@ static bool containsStage(WGPUShaderStageFlags stageBitfield, auto stage)
         if (!descriptor)
```

Comment:
Personally I would prefer not to have the concept of external textures in the bind group layout and have the breakout be performed prior to calling `Device::createBindGroupLayout`. It makes this code path simpler and avoids extending `WGPUBindGroupLayoutEntry`

But indeed this was not changed by this patch.

Reply:
> I would prefer not to have the concept of external textures in the bind group layout and have the breakout be performed prior

But external textures include things which are not resources (clip rect, etc.). So they cannot be decomposed ahead-of-time, and therefore need to be presented as their own item, as a sibling to buffers and textures.

Reply:
External textures don't contain variable information from the argument encoder's perspective, so why can't we decompose them ahead of time?

Reply:
Because each of the individual pieces of an external texture cannot be represented by a `WGPUBindGroupLayoutEntry`. So there's no way we can decompose the external texture above the API boundary and also represent all its constituent pieces. So, instead, I think we have to decompose below the API boundary, and have the API just treat external textures as just another kind of resource.

This is how the W3C's WebGPU API is designed - textures, samplers, buffers, and external textures are all just sibling resources.

Reply:
Filed https://bugs.webkit.org/show_bug.cgi?id=257978 and added a FIXME to track this issue